### ISSUE TITLE
python setup.py fails with python 3.9.2

### ISSUE THREAD

**Issue by erachen** [HybirdCorp/creme_crm#288]
Using debian bullseye 
Step to reproduce:
- create a virtual environnement: `virtualenv --system-site-packages --python=python3 venv`
- use it: `. venv/bin/activate`
- install creme: `pip install 'git+https://github.com/HybirdCorp/creme_crm.git@2.2.4#egg=creme'`

It seems that even if django appear in `install_required`, `setuptools` try to import `creme` and so `creme/__init__.py` that import `django`.

Blaming against `creme/__init__.py` it seems more likely that `setuptools` change it behaviour rather than `creme`.

workaround: installing manually django `pip install 'django ~=2.2.24'`

complete traceback:
```
Collecting creme
  Cloning https://github.com/HybirdCorp/creme_crm.git (to revision 2.2.4) to ./pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5
  Running command git clone -q https://github.com/HybirdCorp/creme_crm.git /tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5
  Running command git checkout -q 46e22576a1be9c0bcd74d38912f04a35fcdd48b5
    ERROR: Command errored out with exit status 1:
     command: /tmp/venv/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5/setup.py'"'"'; __file__='"'"'/tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-qza7ppwn
         cwd: /tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5/
    Complete output (35 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5/setup.py", line 3, in <module>
        setup()
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/__init__.py", line 162, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.9/distutils/core.py", line 121, in setup
        dist.parse_config_files()
      File "/tmp/venv/lib/python3.9/site-packages/_virtualenv.py", line 21, in parse_config_files
        result = old_parse_config_files(self, *args, **kwargs)
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/dist.py", line 701, in parse_config_files
        parse_configuration(self, self.command_options,
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 121, in parse_configuration
        meta.parse()
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 426, in parse
        section_parser_method(section_options)
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 399, in parse_section
        self[name] = value
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 184, in __setitem__
        value = parser(value)
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 515, in _parse_version
        version = self._parse_attr(value, self.package_dir)
      File "/tmp/venv/lib/python3.9/site-packages/setuptools/config.py", line 349, in _parse_attr
        module = import_module(module_name)
      File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
      File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 790, in exec_module
      File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
      File "/tmp/pip-install-o4sneg_g/creme_4b4603647fe34c7fb4d6d282590568b5/creme/__init__.py", line 7, in <module>
        from django.apps.config import AppConfig
    ModuleNotFoundError: No module named 'django'
    ----------------------------------------
WARNING: Discarding git+https://github.com/HybirdCorp/creme_crm.git@2.2.4#egg=creme. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement creme (unavailable)

```
Note: I try with `creme

**Comment by erachen** [comment]
seems to be a `virtualenv` / `setuptools` problem:
- in debian buster `virtualenv` (`15.1`) install `setuptools` (`40.8`) _then_ upgrade it to the last version
- wheras in debian bullseye `virtualenv` (`20.4`) install `setuptools` (`44.1.1`) regardless of `python3-setuptools` (`52.0`) in debian packages and left it alone.

After creating  the  virtualenv, upgrading manually `setuptools` (`pip install --upgrade setuptools`) solve the problem.

There is a bug or at least a change in behavior of `virtualenv` anyway it's not a bug from `creme`.
Sorry for the noise.

**Comment by erachen** [comment]
Build still have issue with python 3.9 since creme required Pillow 7.2.0 and this version is not compatible with python 3.9
cf: https://pillow.readthedocs.io/en/latest/installation.html#python-support

It seems that it is fixed by commit: d9c8e9b064b94cde3dbce21626704b20d5e23550

**Comment by genglert** [comment]
I'll probably will upgrade again the version of some libs before the release of Creme 2.3.
I'll push a fix release of Creme 2.2 with a dependency to a Pillow version which is compatible with Python 3.9 (we do not use Pillow heavily so API breaking is globally not very risky with this lib).
Thx for the report.
